lib/feature_management.rb

```diff
@@ -5,6 +5,12 @@ class FeatureManagement
     idp.pt.identitysandbox.gov
   ].freeze
 
+  ENVS_WHERE_THREATMETRIX_MOCK_NOT_ALLOWED = %w[
+    secure.login.gov
+    idp.staging.login.gov
+    idp.int.login.gov
+  ].freeze
```

Comment:
I am on the fence about needing a list like this. We don't have an equivalent for other vendors? Maybe we need a different, clearer mock config elsewhere? Hardcoding envs like this is not 12-factor-friendly, the GPO example above is not a great pattern IMO

Reply:
I think this is a defensive pattern against accidentally enabling this feature in prod, since that would user-visible and very bad. Agree it's a little smelly. I'd like to make the common case easy (mock enabled locally & in dev) without removing the ability to test locally.

Reply:
yeah I would be a fan of removing the extra layer (because other vendors don't have it), and on the other hand, the fact that it would be user-visible *might* be a good thing because we'd hear about it quickly? 🤷 

anyways, I would love the simplification if we could remove this, but on a scale of 1-10, I'm at a 3 that I think it should be remove, I could easily be convinced to leave it.

Reply:
I ended up rolling this back--I think the magic is maybe not worth it.